auth-api/src/main/java/io/stargate/auth/api/AuthResource.java

```diff
@@ -87,29 +90,83 @@ public Response createToken(Secret body) {
   @Path("/auth")
   @Consumes(MediaType.APPLICATION_JSON)
   @Produces(MediaType.APPLICATION_JSON)
-  public Response createToken(Credentials body) {
-    if (body == null) {
+  public Response createToken(Credentials credentials) {
+    if (credentials == null) {
       return Response.status(Response.Status.BAD_REQUEST)
           .entity(new Error("Must provide a body to the request"))
           .build();
     }
 
-    if (body.getUsername() == null || body.getUsername().equals("")) {
+    if (credentials.getUsername() == null || credentials.getUsername().equals("")) {
       return Response.status(Response.Status.BAD_REQUEST)
           .entity(new Error("Must provide username in request"))
           .build();
     }
 
-    if (body.getPassword() == null || body.getPassword().equals("")) {
+    if (credentials.getPassword() == null || credentials.getPassword().equals("")) {
       return Response.status(Response.Status.BAD_REQUEST)
           .entity(new Error("Must provide password in request"))
           .build();
     }
 
-    // TODO: [doug] 2020-06-18, Thu, 0:42 handle potential errors from authService
     String token;
     try {
-      token = authService.createToken(body.getUsername(), body.getPassword());
+      token = authService.createToken(credentials.getUsername(), credentials.getPassword());
+    } catch (UnauthorizedException e) {
+      return Response.status(Response.Status.UNAUTHORIZED)
+          .entity(new Error("Failed to create token: " + e.getMessage()))
+          .build();
+    } catch (Exception e) {
+      logger.error("Failed to create token", e);
+      return Response.status(Response.Status.INTERNAL_SERVER_ERROR)
+          .entity(new Error("Failed to create token: " + e.getMessage()))
+          .build();
+    }
+
+    return Response.status(Response.Status.CREATED)
+        .entity(new AuthTokenResponse().authToken(token))
+        .build();
+  }
+
+  /**
+   * NOTE: This method is intended to be used in conjunction with another authentication service.
+   * For example, the request has already been authenticated through a proxy but a token is still
+   * needed to make requests to Stargate.
+   *
+   * <p>
+   *
+   * <p>Generates a token based solely on if the provided username exists in the database.
+   *
+   * @param usernameCredentials A username to generate a token for IFF the username exists in the
+   *     database.
+   * @return On success will return a token to be used with later requests to Stargate.
+   */
+  @POST
+  @Path("/auth/usernametoken")
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response createTokenFromUsername(UsernameCredentials usernameCredentials) {
```

Comment:
How do we make sure ordinary users cannot abuse this method? Is it not exposed to the outside traffic? (sorry, I do not know how to check)

Reply:
Hence the note up above. This should be used in a secured manner such as if you already have auth behind a proxy but you still need to generate a token for some username. Also why it's behind a flag that's off by default.